### PR TITLE
Fixes #32937 - Support static networking in AutoYaST iPXE

### PR DESCRIPTION
**Pull request by tux93**

### CODE REVIEW

**Comment by tbrisker** [comment]
ok to test

**Review by tbrisker — COMMENTED** [review]
looks like you accidentally changed the file permissions?

**Comment by tux93** [comment]
Permissions should be fixed now

**Comment by ezr-ondrej** [comment]
@sbernhard would you mind to test this or find someone who can?

**Review by lzap — DISMISSED** [review]
LGTM but I haven't tested it.

File: app/views/unattended/provisioning_templates/iPXE/autoyast_default_ipxe.erb [hunk 1]
```diff
@@ -17,6 +18,11 @@ ping --count 1 ${netX/dns} || echo Ping to DNS failed or ping command not availa
 <% kernel = boot_files_uris[0] -%>
 <% initrd = boot_files_uris[1] -%>
 
+<% subnet = @host.subnet -%>
+<% if subnet.respond_to?(:dhcp_boot_mode?) && subnet.dhcp_boot_mode? -%>
 kernel <%= kernel %> initrd=initrd.img splash=silent install=<%= medium_uri %> autoyast=<%= foreman_url('provision') %> text-mode=1 useDHCP=1
+<% else -%>
+kernel <%= kernel %> initrd=initrd.img splash=silent install=<%= medium_uri %> autoyast=<%= foreman_url('provision') %> text-mode=1 useDHCP=0 netsetup=-dhcp ifcfg=*="${netX/ip}/<%= @host.primary_interface.subnet.cidr -%>,${netX/gateway},${dns},<%= @host.domain %>" "<%= http_proxy_string -%>"
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/iPXE/autoyast_default_ipxe.erb]: Can you move quotes for http_proxy_string out of here to the line 10?

**Comment by lzap** [comment]
Once this is fixed, let's merge, thanks.

File: app/views/unattended/provisioning_templates/iPXE/autoyast_default_ipxe.erb (hunk 1 above)
Comment (tux93) [reply app/views/unattended/provisioning_templates/iPXE/autoyast_default_ipxe.erb]: Done

**Review by sbernhard — DISMISSED** [review]
LGTM. Thanks.

**Comment by ezr-ondrej** [comment]
I'm sorry, but this has conflict now, can you please rebase @tux93 ?

**Comment by tux93** [comment]
> I'm sorry, but this has conflict now, can you please rebase @tux93 ?

Done.

**Review by ezr-ondrej — APPROVED** [review]
Looks ok to me to my best knowledge tho that is not high :)

**Pull request merged**